app/src/main/java/com/radiusnetworks/example/flybuy/GuestJourneyActivity.kt

```diff
@@ -50,42 +51,40 @@ class GuestJourneyActivity : AppCompatActivity() {
     }
 
     private fun orderProgress(order: Order) {
-        // OrderStates of COMPLETED and CANCELLED take priority over CustomerState
         when (order.state) {
-            OrderState.COMPLETED -> showOrderCompleted()
             OrderState.CANCELLED -> startActivity(Intent(this, MainActivity::class.java))
             OrderState.GONE -> startActivity(Intent(this, MainActivity::class.java))
             else -> {
-                when (order.customerState) {
-                    CustomerState.CREATED -> {
-                        progressBar.progress = 25
-                        eta.text = formatETA(order.etaAt)
-                        claimOrder(order)
-                    }
-                    CustomerState.EN_ROUTE -> {
-                        progressBar.progress = 33
-                        eta.text = formatETA(order.etaAt)
-                    }
-                    CustomerState.NEARBY -> {
-                        progressBar.progress = 50
-                        eta.text = formatETA(order.etaAt)
-                    }
-                    CustomerState.ARRIVED -> {
-                        progressBar.progress = 67
-                        textView.text = getString(R.string.youre_here)
-                        eta.text = order.site.instructions
-                    }
-                    CustomerState.WAITING -> {
-                        progressBar.progress = 75
-                        // Once a customer is waiting, hide the I'm here button and show I'm done
-                        textView.text = getString(R.string.youre_here)
-                        eta.text = order.site.instructions
-                        imDoneButton.visibility = View.VISIBLE
-                        imHereButton.visibility = View.INVISIBLE
-                    }
-                    CustomerState.COMPLETED -> {
-                        progressBar.progress = 100
-                        showOrderCompleted()
+                if (!order.open()) {
+                    showOrderCompleted()
+                } else {
+                    when (order.customerState) {
+                        CustomerState.CREATED -> {
+                            progressBar.progress = 25
+                            eta.text = formatETA(order.etaAt)
+                            claimOrder(order)
```

Comment:
This isn't correct here. The customer state could be in `created` for an already claimed order. See the logic above for deciding when to claim the order.